pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -255,6 +260,91 @@ func (n *NetworkPolicyController) deleteNamespace(old interface{}) {
 	}
 }
 
+func (c *NetworkPolicyController) filterAGsFromNodeLabels(node *v1.Node) sets.String {
+	ags := sets.NewString()
+	addressGroupObjs, _ := c.addressGroupStore.GetByIndex(store.IsNodeAddressGroupIndex, store.IsNodeAddressGroupIndex)
+	for _, addressGroupObj := range addressGroupObjs {
+		addressGroup := addressGroupObj.(*antreatypes.AddressGroup)
+		nodeSelector := addressGroup.Selector.NodeSelector
+		if nodeSelector.Matches(labels.Set(node.GetLabels())) {
+			ags.Insert(addressGroup.Name)
+		}
+	}
+	return ags
+}
+
+func (c *NetworkPolicyController) addNode(obj interface{}) {
+	node := obj.(*v1.Node)
+	affectedAGs := c.filterAGsFromNodeLabels(node)
+	for _, key := range affectedAGs.List() {
+		c.enqueueAddressGroup(key)
+	}
+	klog.V(2).InfoS("Processed Node CREATE event", "nodeName", node.Name, "affectedAGs", affectedAGs.Len())
+}
+
+func (c *NetworkPolicyController) deleteNode(obj interface{}) {
+	node, ok := obj.(*v1.Node)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.ErrorS(nil, "Processed Node DELETE event error", "obj", obj)
+			return
+		}
+		node, ok = tombstone.Obj.(*v1.Node)
+		if !ok {
+			klog.ErrorS(nil, "Processed Node DELETE event error", "obj", tombstone.Obj)
+			return
+		}
+	}
+	// enqueue affected address group
+	affectedAGs := c.filterAGsFromNodeLabels(node)
+	for _, key := range affectedAGs.List() {
+		c.enqueueAddressGroup(key)
+	}
+	klog.V(2).InfoS("Processed Node DELETE event", "nodeName", node.Name, "affectedAGs", affectedAGs.Len())
+}
+
+func nodeIPChanged(oldNode, newNode *v1.Node) (changed bool, err error) {
+	var oldIPs, newIPs, oldNodeGWIPs, newNodeGWIPs *ip.DualStackIPs
+	if oldIPs, err = k8s.GetNodeAddrs(oldNode); err != nil {
+		return
+	}
+	if newIPs, err = k8s.GetNodeAddrs(newNode); err != nil {
+		return
+	}
```

Comment:
It should return early if change is already identified. 
And I think if there is an error, we should return changed as true. Imagine this scenario:
A Node doesn't have a valid Node IP because of a bug or it's not reported yet, after it gets an valid IP, why we don't reconcile the policy with the new IP?

Reply:
Good catch. Updated.

Reply:
Second thought, which might make more sense:
We don't need to care whether there is an error when getting oldIPs and newIPs. We should just focus on whether there is a change. If both get error, oldIPs and newIPs are both nil, there is no point to reconcile the rule as well. So the code could just be:
```
oldIPs, _ = k8s.GetNodeAddrs(oldNode)
newIPs, _ = k8s.GetNodeAddrs(newNode)
!reflect.DeepEqual(newIPs, oldIPs) {
    return true
}
```

Reply:
Updated.